packages/gatsby-theme-bodiless/src/dist/GatsbyMobxStoreItem.ts

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright © 2019 Johnson & Johnson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import path from 'path';
+import {
+  observable, action, reaction, IReactionDisposer,
+} from 'mobx';
+// eslint-disable-next-line import/no-cycle
+import GatsbyMobxStore from './GatsbyMobxStore';
+import { ItemStateEvent } from './types';
+
+type MetaData = {
+  author: string;
+};
+
+enum ItemState {
+  Clean,
+  Dirty,
+  Flushing,
+  Locked,
+}
+
+export default class GatsbyMobxStoreItem {
+  @observable data = {};
+
+  @observable state: ItemState = ItemState.Clean;
+
+  isDeleted = false;
```

Comment:
Shouldn't this also be observable? We need a way to notify observers that the item has been deleted. It's probably not an issue now, bc we only delete items which don't have an active reference, but later we may want to delete items which are still visible (eg to revert to default content).

One option here is to create a @computed data getter method which respects `isDeleted` and returns empty data if the item has been deleted.

Reply:
@wodenx, it makes sense to make isDeleted as observable, but tbh I would defer it until the case when it is needed unless we have a use case know in which we can test it